app/helpers/document_helper.rb

```diff
@@ -13,6 +13,14 @@ def evidence_tag?(tag)
     Document::EVIDENCE_TAGS.include?(tag)
   end
 
+  def supporting_document_tag?(tag)
+    Document::SUPPORTING_DOCUMENT_TAGS.include?(tag)
+  end
+
+  def irrelevant_tags(planning_application, key)
+    Document.const_get("#{key.singularize.upcase}_TAGS") - planning_application.application_type.document_tags[key]
+  end
```

Comment:
Better to use a `case` statement here, e.g.

``` ruby
def irrelevant_tags(planning_application, key)
  document_tags = planning_application.application_type.document_tags[key]

  case key
  when "plans"
    Document::PLAN_TAGS - document_tags
  when "evidence"
    Document::EVIDENCE_TAGS - document_tags
  when "supporting_documents"
    Document::SUPPORTING_DOCUMENTS - document_tags
  else
    raise ArgumentError, "Unexpected document tag type: #{key}"
  end
end
```

That way it's obvious where any error is from the Appsignal notification

Reply:
Maybe better to place this method on the model: ``` ruby
class ApplicationType < ApplicationRecord
  def irrelevant_tags(key)
    case key
    when "plans"
      Document::PLAN_TAGS - document_tags[key]
    when "evidence"
      Document::EVIDENCE_TAGS - document_tags[key]
    when "supporting_documents"
      Document::SUPPORTING_DOCUMENTS - document_tags[key]
    else
      raise ArgumentError, "Unexpected document tag type: #{key}"
    end
  end
end
```

Reply:
yep that looks much better